### PR TITLE
Replace intval() function by using direct type casting to (int) where no default value is needed

### PR DESCRIPTION
**Pull request by mhauri**
### Description

Direct type casting is **up to 6x faster** than using casting functions like` intval()`. So this pull request replaces the `intval()` functions by direct type casting to `(int)` where no default value is needed.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] [All automated tests passed successfully (all builds on Travis CI are green)](https://travis-ci.org/mhauri/magento2/builds/397221306)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @mhauri. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: app/code/Magento/Backend/Block/Widget/Button/ButtonList.php [hunk 1]
```diff
@@ -127,8 +127,8 @@ public function getItems()
      */
     public function sortButtons(Item $itemA, Item $itemB)
     {
-        $sortOrderA = intval($itemA->getSortOrder());
-        $sortOrderB = intval($itemB->getSortOrder());
+        $sortOrderA = (int) $itemA->getSortOrder();
```
Comment (orlangur) [inline app/code/Magento/Backend/Block/Widget/Button/ButtonList.php:130]: Please do an amend commit removing spaces, i.e. `(int) $itemA->getSortOrder()` -> `(int)$itemA->getSortOrder()`.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Backend/Block/Widget/Button/ButtonList.php (hunk 1 above)
Comment (orlangur) [reply app/code/Magento/Backend/Block/Widget/Button/ButtonList.php:130]: Will do it on our side as `php-cs-fixer` is able to do it.

Comment (mhauri) [reply app/code/Magento/Backend/Block/Widget/Button/ButtonList.php:130]: @orlangur so I do nothing correct, or should I still do the amend commit?

Comment (orlangur) [reply app/code/Magento/Backend/Block/Widget/Button/ButtonList.php:130]: @mhauri do nothing please, sorry for confusing)

**Comment by mhauri** [comment]
Hi @orlangur in case it helps to process this PR I can still remove the spaces. Let me know if I can help.

**Comment by orlangur** [comment]
@mhauri no need in that, thanks for assistance. Just that I didn't have access to VPN for some time.

**Comment by sidolov** [comment]
Hi @mhauri , great finding! It would be great if you implement sniff for the static check to avoid usage of intval() function in future. As an example you can view sniff implementation for is_null function calls: `\Magento\Sniffs\MicroOptimizations\IsNullSniff`

**Comment by orlangur** [comment]
@sidolov such fixer is already a part of `php-cs-fixer`, so, it will be easier to integrate it as a whole.

Another possible approach - simply expand `ForbiddenFunction` PHPCS sniff.

**Comment by sidolov** [comment]
Hi @mhauri , @orlangur , related pull requests have already merged, so, I think we can proceed with current PR. I propose to implement the PHPCS rule for all forbidden functions. 
It's a really simple way, we just need to add a fragment with forbidden functions to `dev/tests/static/framework/Magento/ruleset.xml`:

```
<rule ref="Generic.PHP.ForbiddenFunctions">
    <properties>
        <property name="forbiddenFunctions" type="array" value="strval=>(string)$var"/>     
    </properties>
</rule>
```

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-3020](https://jira.corp.magento.com/browse/ENGCOM-3020) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @mhauri. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.8. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Pull request merged**